addons/hr/populate/hr.py

```diff
@@ -101,23 +135,35 @@ def get_tag_ids(values, counter, random):
 
         return [
             ('active', populate.iterate([True, False], [0.9, 0.1])),
-            ('company_id', populate.constant(company_id)),
             ('name', populate.constant("employee_{counter}")),
+            ('_user_and_company', _compute_user_and_company),
             ('department_id', populate.randomize(department_ids)),
             ('job_id', populate.randomize(job_ids)),
             ('work_location_id', populate.randomize(work_location_ids)),
             ('category_ids', populate.compute(get_tag_ids)),
+            ('resource_calendar_id', populate.compute(get_resource_calendar_id)),
         ]
 
     def _populate_set_manager(self, employees):
-        manager_ids = []
+        _logger.info("Setting up employee managers")
+        manager_ids = defaultdict(list)
         rand = populate.Random('hr.employee+manager_generator')
 
         for employee in employees:
-            if rand.random() >= 0.85: # 15% of employees are managers
-                manager_ids.append(employee.id)
+            if rand.random() >= 0.85:  # 15% of employees are managers
+                manager_ids[employee.company_id].append(employee.id)
 
         for employee in employees:
-            manager = rand.choice(manager_ids)
+            manager = rand.choice(manager_ids[employee.company_id])
             if manager != employee.id:
                 employee.parent_id = manager
+
+    def _populate_set_users_companies(self, employees):
+        _logger.info("Updating users company_ids")
+        for employee in employees:
+            user_id = employee.user_id.id
+            if user_id:
+                self.env["res.users"].write({
+                    "company_ids": [employee.company_id.id],
+                    "id": user_id
+                })
```

Comment:
That will not work :smile: 

Either you browse() the record based on its id, or even better:

```py
employee.user_id.write({
    # ...
})
```